__tests__/integration/mint-tokens.test.js

```diff
@@ -255,25 +266,89 @@ describe('mint token', () => {
     expect(p2pkh.address.base58).toEqual(address0);
   });
 
+  it('should mint and create data outputs', async () => {
+    const response = await TestUtils.request
+      .post('/wallet/mint-tokens')
+      .send({
+        token: tokenA.uid,
+        amount: 100,
+        data: [ 'foobar1', 'foobar2' ],
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    // Current funds: 9 HTR + 960 TKA
+
+    const transaction = response.body;
+    console.log(JSON.stringify(transaction));
+    expect(transaction.success).toBe(true);
+    // If unshiftData is not specified, the data output will be the last output
+    const dataOutput1 = transaction.outputs[transaction.outputs.length - 2];
+    const dataOutput2 = transaction.outputs[transaction.outputs.length - 1];
+
+    await TestUtils.waitForTxReceived(wallet1.walletId, response.body.hash);
+
+    console.log(`Data output foobar1: ${JSON.stringify(dataOutput1)}`);
+    console.log(`Data output foobar2: ${JSON.stringify(dataOutput2)}`);
+    const script1 = Array.from((new ScriptData('foobar1')).createScript());
+    const script2 = Array.from((new ScriptData('foobar2')).createScript());
+
+    expect(dataOutput1.token_data).toBe(0);
+    expect(dataOutput1.value).toBe(1);
+    expect(dataOutput1.script.data).toEqual(script1);
+
+    expect(dataOutput2.token_data).toBe(0);
+    expect(dataOutput2.value).toBe(1);
+    expect(dataOutput2.script.data).toEqual(script2);
+  });
+
+  it('should mint and create a data output at first position', async () => {
+    const response = await TestUtils.request
+      .post('/wallet/mint-tokens')
+      .send({
+        token: tokenA.uid,
+        amount: 100,
+        data: [ 'foobar' ],
+        unshiftData: true,
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    // Current funds: 9 HTR + 960 TKA
+
+    const transaction = response.body;
+    console.log(JSON.stringify(transaction));
+    expect(transaction.success).toBe(true);
+    const dataOutput = transaction.outputs[0];
+    const script = Array.from((new ScriptData('foobar')).createScript());
+
+    await TestUtils.waitForTxReceived(wallet1.walletId, response.body.hash);
+
+    console.log(JSON.stringify(dataOutput));
+    expect(dataOutput.token_data).toBe(0);
+    expect(dataOutput.value).toBe(1);
+    expect(dataOutput.script.data).toEqual(script);
+  });
+
   it('should mint allowing external authority address', async () => {
+    // XXX: This test should be the last test since it sends the mint authority to the burn address
     const externalAddress = TestUtils.getBurnAddress();
     const response = await TestUtils.request
       .post('/wallet/mint-tokens')
       .send({
         token: tokenA.uid,
-        address: await wallet1.getAddressAt(17),
+        address: await wallet1.getAddressAt(20),
```

Comment:
The 17 address is being used in the tests above since we use address 16 the next available address would be 17, so I needed to push the chosen address here.